app/containers/AccountProvider/saga.js

```diff
@@ -165,6 +183,62 @@ export function* isAvailableAction(isValid) {
   yield call(isValid);
 }
 
+export const getReferralInfo = async (user, eosService) => {
+  const info = await eosService.getTableRow(
+    INVITED_USERS_TABLE,
+    INVITED_USERS_SCOPE,
+    user,
+    process.env.EOS_TOKEN_CONTRACT_ACCOUNT,
+  );
+  return info;
+};
+
+function* updateRefer(user, eosService) {
+  const receivedCookieName = `${REFERRAL_REWARD_RECEIVED}_${user}`;
+  const noInviterCookieName = `${NO_REFERRAL_INVITER}_${user}`;
+  if (getCookie(receivedCookieName) || getCookie(noInviterCookieName)) {
+    return;
+  }
+
+  const info = yield call(getReferralInfo, user, eosService);
+
+  if (info) {
+    const reward = +convertPeerValueToNumberValue(info.common_reward);
+    if (reward) {
+      const locale = yield select(makeSelectLocale());
+      setCookie({ name: receivedCookieName, value: true });
+      yield put(
+        addToast({
+          type: 'success',
+          text: translationMessages[locale][commonMessages.receivedReward.id],
+        }),
+      );
+    }
+  } else {
+    setCookie({
+      name: noInviterCookieName,
+      value: true,
+    });
+  }
+}
+
+const rewardRefer = async (user, eosService) => {
+  try {
+    await eosService.sendTransaction(
+      user,
+      REWARD_REFER,
+      {
+        invited_user: user,
+      },
+      process.env.EOS_TOKEN_CONTRACT_ACCOUNT,
+      null,
+    );
+    return true;
+  } catch (err) {
```

Comment:
This error also needs to be logged. Otherwise we can not see why it fails.

Reply:
Ok, done